django/db/backends/base/base.py

```diff
@@ -501,3 +541,22 @@ def schema_editor(self, *args, **kwargs):
             raise NotImplementedError(
                 'The SchemaEditorClass attribute of this database wrapper is still None')
         return self.SchemaEditorClass(self, *args, **kwargs)
+
+    def on_commit(self, func):
+        if self.in_atomic_block:
+            # transaction in progress; save for execution on commit
+            self.run_on_commit.append((self.savepoint_ids[:], func))
```

Comment:
This will become a performance issue for the database before it becomes one for the Python process :-)